test/commands/source/retrieve.test.ts

```diff
@@ -266,4 +271,34 @@ describe('force:source:retrieve', () => {
     expect(displayStub.calledOnce).to.equal(false);
     expect(getJsonStub.calledOnce).to.equal(true);
   });
+
+  it('should warn users when retrieving CustomField with --metadata', async () => {
+    const metadata = 'CustomField:CustomObject__c.CustomField__c';
```

Comment:
this -m should not produce the warning

`-m CustomField` should produce the warning

`-m CustomField,CustomObject` should not produce the warning